thirdeye/thirdeye-frontend/app/pods/components/range-pill-selectors/component.js

```diff
@@ -0,0 +1,125 @@
+    'Last 7 days': [moment().subtract(6, 'days').startOf('day'), moment().startOf('day').add(1, 'days')],
+    'Last 14 days': [moment().subtract(13, 'days').startOf('day'), moment().startOf('day').add(1, 'days')],
+    'Last 28 days': [moment().subtract(27, 'days').startOf('day'), moment().startOf('day').add(1, 'days')]
+  },
+
+  /**
+   * Reset all time range options and activate the selected one
+   * @method newTimeRangeOptions
+   * @param {String} activeKey - label for currently active time range
+   * @return {undefined}
+   */
+  newTimeRangeOptions(activeKey, start, end) {
+    const timeRangeOptions = this.get('timeRangeOptions');
+
+    // Generate a fresh new range opitons array - all inactive
+    const newOptions = timeRangeOptions.map((range) => {
+      return {
+        name: range.name,
+        value: range.value,
+        isActive: false,
+        start,
+        end
+      };
+    });
+
+    // Activate the last clicked option
+    const foundRangeOption = newOptions.find((range) => range.value === activeKey);
+    if (foundRangeOption) {
+      foundRangeOption.isActive = true;
+    }
+
+    return newOptions;
+  },
+
+  actions: {
+    /**
+     * Invokes the passed selectAction closure action
+     */
+    selectAction(rangeObj) {
+      const action = this.attrs.selectAction;
+      if (action) {
+        return action(rangeObj);
+      }
+    },
+
+    /**
+     * User applies a custom date in date-range-picker, which returns start/end.
+     * Highlight our 'custom' option and bubble it to the controller's 'selectAction'.
+     * @param {String} start - start date
+     * @param {String} end - end date
+     */
+    async onRangeSelection(start, end) {
+      const toggledOptions = this.newTimeRangeOptions('custom', start, end);
+      const customOption = toggledOptions.find(op => op.value === 'custom');
+      this.set('timeRangeOptions', toggledOptions);
+      await this.send('selectAction', customOption);
+    },
+
+    /**
+     * User clicks on a pre-set time range pill link. Highlight link and set active range in date-range-picker.
+     * @param {Object} rangeOption - the selected range object
+     */
+    async onRangeOptionClick(rangeOption) {
+      const { value, start, isActive } = rangeOption;
+      // Handle as a 'range click' only if inactive and not a custom range
+      if (value !== 'custom' && !isActive) {
+        // Set date picker defaults to new start/end dates
+        this.setProperties({
+          activeRangeStart: moment(rangeOption.start).format(RANGE_FORMAT),
+          activeRangeEnd: moment(DEFAULT_END_DATE).format(RANGE_FORMAT)
+        });
+        // Reset options and highlight selected one. Bubble selection to parent controller.
+        this.set('timeRangeOptions', this.newTimeRangeOptions(value, start, DEFAULT_END_DATE));
+        await this.send('selectAction', rangeOption);
```

Comment:
Should these be `sendAction`?

Reply:
Oh i see it's a local action that invokes the actual closure action. OK. Makes sense. Thanks.